app/controllers/application_controller.rb

```diff
@@ -3,19 +3,26 @@ class ApplicationController < ActionController::Base
 
   helper_method :logged_in?, :current_user # Give views access to these application controller methods.
 
+  def require_user
+    if !logged_in?
+      flash[:error] = "You must be logged in to do that."
+      redirect_to front_path
+    end
+  end
+
   # Returns a user object or nil.
   def current_user
     if logged_in?
-      User.find(session[:id])
+      User.find(session[:user_id])
```

Comment:
This will blow up if there is a user_id set but it is the wrong id (theoretically this shouldn't happen, but things happen. Most of the time this validation is done by simply checking if there is a user_id key in the session. Alternatively you could use find_by.